src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,121 @@
+
+import { calcStartPosition } from "./Schedule.utils";
+
+import { ScheduleRoomEvents } from "../ScheduleRoomEvents";
+
+import "./Schedule.scss";
+
+const MAX_SCHEDULE_START_HOUR = 16;
+const MAX_HOUR = 24;
+const MIDDAY_HOUR = 12;
+
+export const Schedule: React.FC<ScheduleProps> = ({ scheduleDay }) => {
+  const getStartHour = useCallback(
+    (utcSeconds: number) => {
+      return utcSeconds >= scheduleDay.dayStartUtcSeconds
+        ? Number(format(fromUnixTime(utcSeconds), "H"))
+        : 0;
+    },
+    [scheduleDay]
+  );
+
+  const roomStartHour = useCallback(
+    (room: RoomWithEvents) =>
+      room.events?.reduce(
+        (acc, event) => Math.min(acc, getStartHour(event.start_utc_seconds)),
+        MAX_SCHEDULE_START_HOUR
+      ) ?? MAX_SCHEDULE_START_HOUR,
+    [getStartHour]
+  );
+
+  const scheduleStartHour = useMemo(
+    () =>
+      Math.min(
+        ...scheduleDay.rooms.map((room) => roomStartHour(room)),
+        MAX_SCHEDULE_START_HOUR
+      ),
+    [scheduleDay, roomStartHour]
+  );
+
+  const hours = useMemo(
+    () =>
+      range(scheduleStartHour, MAX_HOUR).map(
+        (hour: number) =>
+          `${hour % MIDDAY_HOUR || MIDDAY_HOUR} ${
+            hour >= MIDDAY_HOUR ? "PM" : "AM"
+          }`
+      ),
+    [scheduleStartHour]
+  );
+
+  const containerVars = useCss({
+    "--room-count": scheduleDay.rooms.length + 1,
+    "--current-time--position": calcStartPosition(
+      Math.floor(getUnixTime(Date.now())),
+      scheduleStartHour
+    ),
+    "--hours-count": hours.length,
+  });
+
+  const containerClasses = useMemo(
+    () => classNames("ScheduledEvents", containerVars),
+    [containerVars]
+  );
+
+  return (
+    <div className={containerClasses}>
+      <div className="ScheduledEvents__rooms">
+        <div className="ScheduledEvents__room">
+          <p className="ScheduledEvents__room-title">My Daily Schedule</p>
+          <span className="ScheduledEvents__events-count">
+            {scheduleDay.personalEvents.length} events
+          </span>
+        </div>
+        {scheduleDay.rooms.map((room) => (
+          <div key={room.title} className="ScheduledEvents__room">
+            <p className="ScheduledEvents__room-title">{room.title}</p>
+            <span className="ScheduledEvents__events-count">
+              {room.events?.length || 0} events
+            </span>
+          </div>
+        ))}
+      </div>
+
+      <div className="ScheduledEvents__schedule">
+        <div className="ScheduledEvents__timeline">
+          {hours.map((hour: string) => (
+            <span className="ScheduledEvents__hour" key={hour}>
+              {hour}
+            </span>
+          ))}
+        </div>
+
+        {scheduleDay.isToday && (
+          <div className="ScheduledEvents__current-time-line"></div>
+        )}
+
+        <div className="ScheduledEvents__user-schedule">
+          <ScheduleRoomEvents
+            isUsers
```

Comment:
Used in `src/components/molecules/ScheduleEvent/ScheduleEvent.tsx` here:
![image](https://user-images.githubusercontent.com/753891/117592125-c5f18b80-b17a-11eb-80ff-92754ed14142.png)


Still not sure of its purpose though. Can almost certainly have a better name I expect.

Reply:
hmm not sure what name would be better here, renamed to `isPersonalizedEvent`: https://github.com/sparkletown/sparkle/pull/1302/commits/3773b544eaa9b24083615b54b444167ca724eff2
it means: is the event in the 1st line, where all the saved events are displayed, it's needed for the styling. The background, color of the personal events are different.
![image](https://user-images.githubusercontent.com/63313732/118137894-b9b43b00-b3d3-11eb-889a-45d5814a318a.png)

Reply:
Makes sense. I would probably name the component prop `personalisedEvent`, then rename it in the extract as `isPersonalisedEvent`

```typescript
export interface ScheduleEventProps {
 // ..snip..
  personalizedEvent?: boolean;
}

export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
  event,
  scheduleStartHour,
  isUsers,
  personalizedEvent: isPersonalizedEvent,
}) => {

// ..snip..
```

That way when we call the component it will be used like:

```typescript
<ScheduleEvent /* other props here*/ personalizedEvent>
```

Reply:
done here: https://github.com/sparkletown/sparkle/pull/1302/commits/84c633a2848faab78da90654662b3d1b8aba1942